flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1060,21 +1061,73 @@ private static Function instantiateHiveFunction(ObjectPath functionPath, HiveCat
 		);
 	}
 
+	private boolean isTablePartitioned(Table hiveTable) {
+		return hiveTable.getPartitionKeysSize() != 0;
+	}
+
 	// ------ stats ------
 
 	@Override
 	public void alterTableStatistics(ObjectPath tablePath, CatalogTableStatistics tableStatistics, boolean ignoreIfNotExists) throws TableNotExistException, CatalogException {
-
+		try {
+			Table hiveTable = getHiveTable(tablePath);
+			// Set table stats
+			if (needToUpdateStatistics(tableStatistics, hiveTable.getParameters())) {
+				updateStatisticsParameters(tableStatistics, hiveTable.getParameters());
+				client.alter_table(tablePath.getDatabaseName(), tablePath.getObjectName(), hiveTable);
+			}
+		} catch (TableNotExistException e) {
+			if (!ignoreIfNotExists) {
+				throw e;
+			}
+		} catch (TException e) {
+			throw new CatalogException(String.format("Failed to alter table stats of table %s", tablePath.getFullName()), e);
+		}
 	}
 
 	@Override
 	public void alterTableColumnStatistics(ObjectPath tablePath, CatalogColumnStatistics columnStatistics, boolean ignoreIfNotExists) throws TableNotExistException, CatalogException {
 
 	}
 
+	private static boolean needToUpdateStatistics(CatalogTableStatistics statistics, Map<String, String> oldParameters) {
+		String oldRowCount = oldParameters.getOrDefault(StatsSetupConst.ROW_COUNT, "0");
```

Comment:
These are basically setting default values for all types of Hive table stats. How about moving the `0`s to `HiveStatsUtil` and define them as `public static final` vars to make it easier to find and access across hive catalog code base?

Reply:
Why don't we need to check `CatalogTableStatistics::properties`? Do we allow user to alter what's inside `CatalogTableStatistics::properties`?

Reply:
@zjuwangg can you address these comments?

Reply:
> CatalogTableStatistics

Now we alter hive table stats just consider specific four stats const and don't use `CatalogTableStatistics::properties`.